File: data/readable_format/cpp/file883.cpp
#include <string>
#include <array>

// Equivalent of the COMMON block
struct LIN_DATA {
    std::array<std::string, 4> LIN_ERRMSG;

    LIN_DATA() : LIN_ERRMSG{ "Success",
                             "Null linprm pointer passed",
                             "Memory allocation failed",
                             "PCi_ja matrix is singular" } {}
};

// Global instance
LIN_DATA lin_data;

int main() {
    // Example usage
    for (const auto& msg : lin_data.LIN_ERRMSG) {
        std::cout << msg << std::endl;
    }

    return 0;
}

File: data/readable_format/fortran/file883.f90
PROGRAM testLinErrMsg
    CHARACTER(80) LIN_ERRMSG(0:3)
    COMMON /LIN_DATA/ LIN_ERRMSG

    LIN_ERRMSG(0) = 'Success'
    LIN_ERRMSG(1) = 'Null linprm pointer passed'
    LIN_ERRMSG(2) = 'Memory allocation failed'
    LIN_ERRMSG(3) = 'PCi_ja matrix is singular'

    CALL CheckErrMsg(0, 'Success')
    CALL CheckErrMsg(1, 'Null linprm pointer passed')
    CALL CheckErrMsg(2, 'Memory allocation failed')
    CALL CheckErrMsg(3, 'PCi_ja matrix is singular')

    STOP
END PROGRAM testLinErrMsg

SUBROUTINE CheckErrMsg(index, expected)
    INTEGER index
    CHARACTER*(*) expected
    CHARACTER(80) LIN_ERRMSG(0:3)
    COMMON /LIN_DATA/ LIN_ERRMSG

    IF (LIN_ERRMSG(index) .EQ. expected) THEN
        PRINT *, 'Test passed for index:', index
    ELSE
        PRINT *, 'Test FAILED for index:', index
        PRINT *, 'Expected:', expected
        PRINT *, 'Got:', LIN_ERRMSG(index)
    END IF
END SUBROUTINE CheckErrMsg